### PR TITLE
Update facesContainer-4.1 to accept 4.1 jars

### PR DESCRIPTION
**Pull request by volosied**
Created `io.openliberty.facesContainer.4.1.internal.classloading` for facesContainer-4.1

Follow up to create tests: https://github.com/OpenLiberty/open-liberty/issues/27415

### CODE REVIEW

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
#build

File: dev/io.openliberty.facesContainer.4.1.internal.classloading/bnd.bnd [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#*******************************************************************************
+# Copyright (c) 2023. 2024 IBM Corporation and others.
```
Comment (pnicolucci) [inline dev/io.openliberty.facesContainer.4.1.internal.classloading/bnd.bnd]: `,` not `.`

File: dev/io.openliberty.facesContainer.4.1.internal.classloading/src/io/openliberty/faces41/internal/container/classloading/Faces41LibraryIntegration.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import com.ibm.wsspi.library.ApplicationExtensionLibrary;
+import com.ibm.wsspi.library.Library;
+
+@Component(name = "com.ibm.ws.jsf.container.classloading.4.1",
+           configurationPolicy = ConfigurationPolicy.IGNORE,
+           immediate = true)
+public class Faces41LibraryIntegration implements ApplicationExtensionLibrary {
+
+    // Defines a required dependency on the <library> config element with the
+    // id="io.openliberty.facesContainer.4.0" (provided by defaultInstances.xml in this same bundle)
```
Comment (pnicolucci) [inline dev/io.openliberty.facesContainer.4.1.internal.classloading/src/io/openliberty/faces41/internal/container/classloading/Faces41LibraryIntegration.java]: 4.1?

Comment (volosied) [reply dev/io.openliberty.facesContainer.4.1.internal.classloading/src/io/openliberty/faces41/internal/container/classloading/Faces41LibraryIntegration.java]: Thanks! Also spotted an error in `defaultInstances.xml`.

Should be good now :)

**Comment by volosied** [comment]
Feature works (myfaces loads fine), but I think we are need updates from the other features.  (java.lang.NoClassDefFoundError: jakarta/enterprise/inject/spi/el/ELAwareBeanManager  occurs) 

Confirms a 4.1 jar must be loaded or an error occurs (tested against 4.0.0 jar bundled in the app): 
```
[AUDIT   ] CWWKT0016I: Web application available (default_host): http://localhost:9080/JSF22FacesFlows_MyFaces/
[ERROR   ] JSFG0103E: The JSF specification API version available to application JSF22FacesFlows_MyFaces is 4.0.0-RC4, but it needs to be within the version range 4.1.*. Ensure that the JSF API that is packaged within the indicated application corresponds to the version of jsfContainer that is enabled. Liberty determines the version of the JSF API by checking the Specification-Version manifest attribute of the .jar file that contains a JSF API class.
JSFG0103E: The JSF specification API version available to application JSF22FacesFlows_MyFaces is 4.0.0-RC4, but it needs to be within the version range 4.1.*. Ensure that the JSF API that is packaged within the indicated application corresponds to the version of jsfContainer that is enabled. Liberty determines the version of the JSF API by checking the Specification-Version manifest attribute of the .jar file that contains a JSF API class.
[ERROR   ] An error occured while initializing MyFaces: JSFG0103E: The JSF specification API version available to application JSF22FacesFlows_MyFaces is 4.0.0-RC4, but it needs to be within the version range 4.1.*. Ensure that the JSF API that is packaged within the indicated application corresponds to the version of jsfContainer that is enabled. Liberty determines the version of the JSF API by checking the Specification-Version manifest attribute of the .jar file that contains a JSF API class.
JSFG0103E: The JSF specification API version available to application JSF22FacesFlows_MyFaces is 4.0.0-RC4, but it needs to be within the version range 4.1.*. Ensure that the JSF API that is packaged within the indicated application corresponds to the version of jsfContainer that is enabled. Liberty determines the version of the JSF API by checking the Specification-Version manifest attribute of the .jar file that contains a JSF API class.
[AUDIT   ] CWWKZ0001I: Application JSF22FacesFlows_MyFaces started in 0.991 seconds.

```

Current error when I try jakarta.faces 4.0.0-M1
```
[1/19/24, 15:18:41:027 EST] 00000049 com.ibm.ws.logging.internal.impl.IncidentImpl                I FFDC1015I: An FFDC Incident has been created: "java.lang.NoClassDefFoundError: jakarta/enterprise/inject/spi/el/ELAwareBeanManager com.ibm.ws.webcontainer.osgi.DynamicVirtualHost startWebApp" at ffdc_24.01.19_15.18.40.0.log
[1/19/24, 15:18:41:029 EST] 00000049 com.ibm.ws.app.manager.AppMessageHelper                      A CWWKZ0020I: Application JSF22FacesFlows_MyFaces not updated.
```

4.1.0-SNAPSHOT MyFaces: 
```
[1/19/24, 15:44:47:824 EST] 0000006c com.ibm.ws.webcontainer.osgi.webapp.WebGroup                 I SRVE0169I: Loading Web Module: JSF22FacesFlows.
[1/19/24, 15:44:47:824 EST] 0000006c com.ibm.ws.webcontainer                                      I SRVE0250I: Web Module JSF22FacesFlows has been bound to default_host.
[1/19/24, 15:44:47:825 EST] 0000006c com.ibm.ws.http.internal.VirtualHostImpl                     A CWWKT0016I: Web application available (default_host): http://localhost:9080/JSF22FacesFlows_MyFaces/
```

**Review by pnicolucci — APPROVED** [review]

**Comment by volosied** [comment]
#libby

**Pull request merged**